Add ability to reconfigure v2 project rules

johnnyd-eth opened the pull request:
## What does this PR do and why?

Tx still not working (we think Rink contracts needs updating), but everything in the rules tab is loading and submitting to the tx properly. 

Other code changes:
- created component for `ProjectConfigurationFieldsContainer` to avoid duplicate `<Col md={!hidePreview ? 10 : 24} xs={24}>` (this, as well as all the tabs should be moved out of v2create and into a new directory in a follow-up PR)
- create generic component for funding/token/rules tabs

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/160156420-e9ca773d-cdbd-49f6-b229-58c3c0b25b4e.mp4

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

aeolianeth commented on src/components/v2/V2Create/FormItemLabel.tsx:
```diff
@@ -1,11 +1,18 @@
+  children,
+  style,
+}: {
+  children: ChildElems | string
```
**issue**: we can keep PropsWithChildren like this:

`PropsWithChildren<{ style?: CSSProperties }>`

johnnyd-eth replied:
Oh ok nice

aeolianeth commented on src/components/v2/V2Create/tabs/ProjectConfigurationFieldsContainer.tsx:
```diff
@@ -0,0 +1,16 @@
+  hidePreview,
+  children,
+}: {
+  hidePreview?: boolean
```
**issue**: lets use `PropsWithChildren` here

aeolianeth commented on src/components/v2/V2Create/tabs/ProjectConfigurationFieldsContainer.tsx:
```diff
@@ -0,0 +1,16 @@
+  children?: ChildElems
+}) {
+  return (
+    <Col md={!hidePreview ? 10 : 24} xs={24}>
```
**praise** yeyuh

aeolianeth commented on src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx:
```diff
@@ -37,22 +43,21 @@ export default function RulesTabContent() {
     form.setFieldsValue({
-      pausePay: fundingCycleMetadata?.pausePay ?? false,
-      pauseMint: fundingCycleMetadata?.pauseMint ?? false,
+      pausePay: fundingCycleMetadata?.pausePay,
```
**question** why this change? does `undefined` produce the same outcome? feel like we could be explicit and keep it as `false`

johnnyd-eth replied:
`undefined` does produce the same outcome

aeolianeth commented on src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx:
```diff
@@ -87,36 +91,29 @@ export default function RulesTabContent() {
-                onChange={val => form.setFieldsValue({ pausePay: val })}
-              />
-            </div>
+            <Switch />
```
haha nice, good catch.

aeolianeth commented on src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureFundingDrawer.tsx:
```diff
@@ -1,32 +1,28 @@
     <Drawer visible={visible} {...drawerStyle} onClose={onClose}>
       <h3>
-        <Trans>Reconfigure funding target/duration</Trans>
+        <Trans>{title}</Trans>
```
**issue**: shouldn't wrap in `Trans` tag if there's no static text

johnnyd-eth replied:
Ahh yep good catch

aeolianeth commented on src/components/v2/V2Project/index.tsx:
```diff
@@ -54,9 +54,10 @@ export default function V2Project() {
 
   const distributionLimitCurrencySymbol =
-    currencyMetadata[distributionLimitCurrency?.toNumber() ?? 1].symbol
+    currencyMetadata[distributionLimitCurrency?.toNumber() ?? 1]?.symbol
```
**issue** lets use `V2CurrencySymbol` function here, and also I don't think we should fall back to ETH - we should probably just let it fail.

johnnyd-eth replied:
Yeah ok

aeolianeth commented on src/locales/ru/messages.po:
```diff
@@ -1440,16 +1440,21 @@ msgstr "Перенастроить финансирование"
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureProjectDetailsDrawer.tsx
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx
 msgid "Reconfigure project details"
+msgstr "<<<<<<< HEAD"
```
**issue**

aeolianeth commented on src/locales/tr/messages.po:
```diff
@@ -1440,16 +1440,21 @@ msgstr "Finansmanı yeniden yapılandır"
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureProjectDetailsDrawer.tsx
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx
 msgid "Reconfigure project details"
+msgstr "<<<<<<< HEAD"
```
**issue**

aeolianeth reviewed (changes requested):
@johnnyd-eth a few comments for ya but on the whole LGTM!

johnnyd-eth commented:
Thanks @aeolianeth I've addressed all those comments. 

Also added some further changes:
- 'Reconfigure upcoming' button on `V2FundingCycleSection` -> uses V2ReconfigureModal but hides projectDetails 
- add 'Owner can token minting' to FC card
- fix bug in `RulesTabContent` where `pauseMint` was getting the opposite value to what was specified in the 'Allow token minting' field

aeolianeth commented on src/utils/v2/currency.ts:
```diff
@@ -30,15 +30,15 @@ const CURRENCY_MAP: { [key in V1CurrencyOption]: V2CurrencyOption } = {
   currency?: V2CurrencyOption,
 ): CurrencyName | undefined =>
-  currency !== undefined ? V2_CURRENCY_METADATA[currency].name : undefined
+  currency !== undefined ? V2_CURRENCY_METADATA[currency]?.name : undefined
```
@johnnyd-eth presumedly if `currency` was defined then `V2_CURRENCY_METADATA[currency]` is always defined. Any particular reason for this change?

aeolianeth reviewed (approved):
@johnnyd-eth apppprroooovvveeeddd
